x-pack/plugins/lens/public/trigger_actions/open_lens_config/create_action_helpers.ts

```diff
@@ -31,20 +32,41 @@ export const [getDatasourceMap, setDatasourceMap] = createGetterSetter<
   Record<string, Datasource<unknown, unknown>>
 >('DatasourceMap', false);
 
-export function isCreateActionCompatible(core: CoreStart) {
-  return core.uiSettings.get(ENABLE_ESQL);
+export async function isCreateActionCompatible(
+  core: CoreStart,
+  editorFrameService: EditorFrameService
+) {
+  let isCompatible: boolean = core.uiSettings.get(ENABLE_ESQL);
+
+  if (isCompatible && !getDatasourceMap() && !getVisualizationMap()) {
+    const [visualizationMap, datasourceMap] = await Promise.all([
+      editorFrameService.loadVisualizations(),
+      editorFrameService.loadDatasources(),
+    ]);
+
+    if (visualizationMap && datasourceMap) {
+      setDatasourceMap(datasourceMap);
+      setVisualizationMap(visualizationMap);
+    } else {
+      isCompatible = false;
+    }
+  }
+
+  return isCompatible;
```

Comment:
Given this PR introduces change to fetch only UIActions registered with the trigger ADD_PANEL_TRIGGER it resulted in required that data dependency that would have previously been made available when the classic lens Embeddable isEditable method is invoked being missing, see [v8.14.3/x-pack/plugins/lens/public/embeddable/embeddable_factory.ts#L70-L73](https://github.com/elastic/kibana/blob/v8.14.3/x-pack/plugins/lens/public/embeddable/embeddable_factory.ts?rgh-link-date=2024-07-23T09%3A28%3A33Z#L70-L73) and [v8.14.3/x-pack/plugins/lens/public/plugin.ts#L386-L389](https://github.com/elastic/kibana/blob/v8.14.3/x-pack/plugins/lens/public/plugin.ts?rgh-link-date=2024-07-23T09%3A28%3A33Z#L386-L389), the above change ensures that the required data dependency is articulated and exists before the action's execute method is invoked.